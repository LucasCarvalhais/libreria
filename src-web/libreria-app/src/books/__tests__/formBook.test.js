import React from 'react';
import { mount } from 'enzyme';
import { FormBook } from '../FormBook';

const handleChange = jest.fn();
const handleSubmit = jest.fn();

const book = { 
    title: 'Teste1', 
    description: 'Teste2', 
    author: 'Teste3', 
    edition: 2, 
};

describe('FormBook', () => {
    const form = mount(
        <FormBook 
            book={book} 
            handleChange={handleChange} 
            handleSubmit={handleSubmit} 
        />
    );
    
    test('should call the function when input changes', () => {
        const input = form.find('input').at(0);
        input.simulate('change');
        expect(handleChange).toHaveBeenCalled();
    });

    test('should have values according of the book', () => {
        expect(form.find('input').at(0).props().value).toBe('Teste1');
        expect(form.find('input').at(1).props().value).toBe('Teste2');
        expect(form.find('input').at(2).props().value).toBe('Teste3');
        expect(form.find('input').at(3).props().value).toBe(2);
    });

    test('should call the function of submit when form submits', () => {
        form.simulate('submit');
        expect(handleSubmit).toHaveBeenCalled();
    });
})