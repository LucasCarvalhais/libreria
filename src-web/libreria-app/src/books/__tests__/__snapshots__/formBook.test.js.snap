// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBook has a valid snapshot 1`] = `
<form
  className="formulario"
  onSubmit={[Function]}
>
  <label>
    <span
      className="legend"
    >
      Título: 
    </span>
    <input
      className="inputForm"
      name="title"
      onChange={[Function]}
      type="text"
      value="Teste"
    />
  </label>
  <br />
  <label>
    <span
      className="legend"
    >
      Descripción: 
    </span>
    <input
      className="inputForm"
      name="description"
      onChange={[Function]}
      type="text"
      value="Teste"
    />
  </label>
  <br />
  <label>
    <span
      className="legend"
    >
      Autor: 
    </span>
    <input
      className="inputForm"
      name="author"
      onChange={[Function]}
      type="text"
      value="teste"
    />
  </label>
  <br />
  <label>
    <span
      className="legend"
    >
      Edición: 
    </span>
    <input
      className="inputForm"
      name="edition"
      onChange={[Function]}
      type="number"
      value={1}
    />
  </label>
  <br />
  <input
    className="submitButton"
    type="submit"
    value="Cadastrar"
  />
</form>
`;
